pkg/controller/studyjob/katib_api_util.go

```diff
@@ -22,41 +22,58 @@ import (
 	katibapi "github.com/kubeflow/katib/pkg/api"
 	katibv1alpha1 "github.com/kubeflow/katib/pkg/api/operators/apis/studyjob/v1alpha1"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
-func initializeStudy(instance *katibv1alpha1.StudyJob, ns string) error {
+func initializeStudy(instance *katibv1alpha1.StudyJob, ns string) (bool, error) {
 	if validErr := validateStudy(instance, ns); validErr != nil {
 		instance.Status.Condition = katibv1alpha1.ConditionFailed
-		return validErr
+		return true, validErr
 	}
 
 	if instance.Spec.SuggestionSpec.SuggestionAlgorithm == "" {
-		instance.Spec.SuggestionSpec.SuggestionAlgorithm = "random"
+		instance.Spec.SuggestionSpec.SuggestionAlgorithm = "manual"
 	}
-
 	instance.Status.Condition = katibv1alpha1.ConditionRunning
-
 	conn, err := grpc.Dial(pkg.ManagerAddr, grpc.WithInsecure())
 	if err != nil {
 		log.Printf("Connect katib manager error %v", err)
 		instance.Status.Condition = katibv1alpha1.ConditionFailed
-		return nil
+		return true, nil
 	}
 	defer conn.Close()
 	c := katibapi.NewManagerClient(conn)
 
-	studyConfig, err := getStudyConf(instance)
-	if err != nil {
-		return err
-	}
-	log.Printf("Create Study %s", studyConfig.Name)
-	//CreateStudy
-	studyID, err := createStudy(c, studyConfig)
-	if err != nil {
-		return err
+	if instance.Spec.ReuseStudyID != "" {
+		_, err = getStudyfromDB(c, instance.Spec.ReuseStudyID)
+		instance.Status.Condition = katibv1alpha1.ConditionFailed
```

Comment:
why set it Failed here

Reply:
Sorry I should move it inside of `if err != nil {`.